routes/tasks.js

```diff
@@ -90,22 +116,51 @@ router
         "title",
         constants.stringLimits["title"]
       );
-      utils.validateStringInputWithMaxLength(
-        textBody,
-        "textBody",
-        constants.stringLimits["textBody"]
-      );
-      utils.validateDate(dateAddedTo, "dateAddedTo");
+
+      if (typeof textBody === "string" && textBody.trim().length > 0) {
+        utils.validateStringInputWithMaxLength(
+          textBody,
+          "textBody",
+          constants.stringLimits["textBody"]
+        );
+        taskPutData.textBody = taskPutData.textBody.trim();
+      } else {
+        taskPutData.textBody = "";
+      }
+      if (typeof dateAddedTo === "string" && dateAddedTo.trim().length > 0) {
+        utils.validateDate(dateAddedTo, "dateAddedTo");
+        taskPutData.dateAddedTo = dayjs(dateAddedTo.trim()).format(
+          "YYYY-MM-DDTHH:mm"
+        );
+      } else {
+        taskPutData.dateAddedTo = "";
```

Comment:
this should be null or empty string ? could you explain if the data layer is expecting a string to decide what to store in the mongo collection ?